src/nvim/buffer_defs.h

```diff
@@ -930,6 +932,10 @@ struct window_S {
 
   synblock_T  *w_s;                 /* for :ownsyntax */
 
+  int w_hl_id;                      ///< 'winhighlight' id
+  int w_hl_id_inactive;             ///< 'winhighlight' id for inactive window
```

Comment:
Actually, if we do `'highlight'` reform, this should just mimic the global `hl_attr()` list. Then we can have `hl_attr_w(wp, HLF_XX)`, support a lot of stuff by doing `:%s/hl_attr(/hl_attr_w(wp,/gc` in screen.c and as a bonus remove a lot of the _ad-hoc_ logic this PR currently needed to deal with vim's _ad-hoc_ logic.